pennylane/tape/tape.py

```diff
@@ -478,6 +480,21 @@ def _update_circuit_info(self):
         self.is_sampled = any(m.return_type is Sample for m in self.measurements)
         self.all_sampled = all(m.return_type is Sample for m in self.measurements)
 
+    def _update_batch_size(self):
+        candidate = None
+        for op in self.operations:
+            op_batch_size = getattr(op, "batch_size", None)
+            if op_batch_size is None:
+                continue
+            if candidate and op_batch_size != candidate:
+                raise ValueError(
+                    "The batch sizes of the tape operations do not match, they include "
+                    f"{candidate} and {op_batch_size}."
+                )
+            candidate = candidate or op_batch_size
```

Comment:
I have a question :) I imagine there will be cases where only _trainable_ parameters/ops have the batch dimension, and non-trainable parameters/operations do not.

E.g., the parameter shift rule:

```python
@qml.qnode(dev)
def circuit(x):
    qml.RY(np.array(0.5, requires_grad=False), wires=0)
    qml.RX(x, wires=0)
    return qml.expval(qml.PauliZ(0))
```

This QNode would create a tape where `trainable_params==[1]`.

If we want to batch the parameter-shift rule, we would have something like this:

```python
qml.RY(0.5)
qml.RX([x+pi/2, x-pi/2)]
```

Reply:
Actually, I'm not sure. We would just do this instead, I guess?

```python
qml.RY([0.5, 0.5])
qml.RX([x+pi/2, x-pi/2)]
```

Reply:
I've confused myself. I guess my original question is: for the batch size validation, do we want to check _all_ ops in the tape, or just those with batch dimensions

Reply:
Nevermind - I reread the logic again, and realized that the loop `continue`'s if the batch dimension is none 🤦 So looks like it works well!

Reply:
I hope so! :) In contrast to `batch_params`, unbatched parameters/operations are kind of "ignored" because they will be broadcast by NumPy/TF/Torch...